src/transformers/models/bridgetower/configuration_bridgetower.py

```diff
@@ -0,0 +1,403 @@
+
+        if config_dict.get("model_type") == "bridgetower":
+            config_dict = config_dict["text_config"]
+
+        if "model_type" in config_dict and hasattr(cls, "model_type") and config_dict["model_type"] != cls.model_type:
+            logger.warning(
+                f"You are using a model of type {config_dict['model_type']} to instantiate a model of type "
+                f"{cls.model_type}. This is not supported for all configurations of models and can yield errors."
+            )
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class BridgeTowerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`BridgeTowerModel`]. It is used to instantiate a
+    BridgeTower model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the bridgetower-base
+    [BridegTower/bridgetower-base](https://huggingface.co/BridgeTower/bridgetower-base/) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        cross_modal_transform_shared (`bool`, *optional*, defaults to `True`):
+            Whether cross modal transformer layers are shared.
+        drop_rate (`float`, *optional*, defaults to 0.1):
+            Drop out probability.
+        head_hidden_scale (`int`, *optional*, defaults to 2):
+            Scale of hidden layers head.
+        hidden_act (`str` or `function`, *optional*, defaults to `"gelu"`):
+            The non-linear activation function (function or string) in the encoder and pooler.
+        hidden_size (`int`, *optional*, defaults to 768):
+            Dimensionality of the encoder layers and the pooler layer.
+        image_size (`int`, *optional*, defaults to 288):
+            The size (resolution) of each image.
+        input_image_embed_size (`int`, *optional*, defaults to 768):
+            Embedding size of the input image.
+        input_text_embed_size (`int`, *optional*, defaults to 768):
+            Embedding size of the input text.
+        is_encoder_decoder (`bool`, *optional*, defaults to `False`):
+            Whether this is an encoder/decoder model
+        layer_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the layer normalization layers.
+        link_tower_shared (`bool`, *optional*, defaults to `False`):
+            Whether the bride/link tower layers are shared.
+        link_tower_type (`str`, *optional*, defaults to `"add"`):
+            Type of the bridge/link layer.
+        max_text_len (`int`, *optional*, defaults to 50):
+            Maximum text length.
+        mlp_ratio (`int`, *optional*, defaults to 4):
+            Ratio of MLP hidden dim to embedding dim.
```

Comment:
```suggestion
            Ratio of MLP hidden dimension to embedding dimension.
```

Reply:
Cleaned up the configs. Thanks for the suggestion!